### DISCUSSION TITLE
How to disable hotkey like ctrl+n?

### DISCUSSION THREAD

**Discussion by winston0410** [Eloston/ungoogled-chromium#1563] [Q&A]
Is it possible to disable hotkey in chromium, like ctrl+n?

**Comment by Ahrotahn** [comment]
Keyboard shortcuts are handled in `chrome/browser/ui/browser_command_controller.cc`.  

You would be able to disable ctrl+n by making this change:
```diff
--- a/chrome/browser/ui/browser_command_controller.cc
+++ b/chrome/browser/ui/browser_command_controller.cc
@@ -1144,7 +1144,7 @@
   IncognitoModePrefs::Availability incognito_availability =
       IncognitoModePrefs::GetAvailability(profile->GetPrefs());
   command_updater->UpdateCommandEnabled(
-      IDC_NEW_WINDOW, incognito_availability != IncognitoModePrefs::FORCED);
+      IDC_NEW_WINDOW, false);
   command_updater->UpdateCommandEnabled(
       IDC_NEW_INCOGNITO_WINDOW,
       incognito_availability != IncognitoModePrefs::DISABLED &&
```

**Comment by winston0410** [comment]
@Ahrotahn thank you for the reply. Are those file inside `chrome/browser/` save to edit? I am not familiar with C development, are those files built or not?

Actually I would like to map it `ctrl+n` to `ArrowDown`, what is my best bet for that? Should I done that in GTK mapping?

**Reply by Ahrotahn** [reply]
Those files need to be changed before building, so this would only work when compiling your own version of chromium.  

If you want the down arrow to open new windows, then you can modify the hotkeys defined in `chrome/browser/ui/views/accelerator_table.cc`:
```diff
--- a/chrome/browser/ui/views/accelerator_table.cc
+++ b/chrome/browser/ui/views/accelerator_table.cc
@@ -181,7 +181,7 @@
     {ui::VKEY_N, ui::EF_SHIFT_DOWN | ui::EF_PLATFORM_ACCELERATOR,
      IDC_NEW_INCOGNITO_WINDOW},
     {ui::VKEY_T, ui::EF_PLATFORM_ACCELERATOR, IDC_NEW_TAB},
-    {ui::VKEY_N, ui::EF_PLATFORM_ACCELERATOR, IDC_NEW_WINDOW},
+    {ui::VKEY_DOWN, ui::EF_NONE, IDC_NEW_WINDOW},
     {ui::VKEY_T, ui::EF_SHIFT_DOWN | ui::EF_PLATFORM_ACCELERATOR,
      IDC_RESTORE_TAB},
 
```

If you're trying to make the down arrow actually return ctrl+n as a key combination, then it may be better to find a method to do this in your system.  Maybe a program like sxhkd, xmodmap, or xbindkeys would work if there isn't an option in your DE to set that.  Those are probably better bets if you don't wish to compile your own version of chromium as well.

**Reply by winston0410** [reply]
@Ahrotahn thanks. I see, then I guess building my own version will be a better version, but where can I find `chrome/browser/ui/views/accelerator_table.cc`? I have searched in the repo, checked `~/.config` and `/etc`, yet I cannot find it.

I am actually trying to make `ctrl+n` to act like `ArrowDown` and `ctrl+p` to act Like `ArrowUp`, so I can navigate the result without touching my arrow keys.

**Reply by Ahrotahn** [reply]
That file is part of the chromium source and not in the repo.  

I believe you'd have an easier time setting up sxhkd.  It would be able to capture the combination without letting it through to the browser, and you'd be able to modify the config as you need without having to spend hours recompiling.  

There is a good writeup on Arch's wiki [here](https://wiki.archlinux.org/title/Sxhkd).  You would only need to install your distro's sxhkd package and either the xautomation or xdotool package.  

To get ctrl+n to produce the same effect as pressing the down arrow your config would have something like:
```
control + n
    xte "keydown Down" "keyup Down"
```

**Reply by winston0410** [reply]
@Ahrotahn thank you for your advice. I will give `sxhkd` a try then, thanks!
